test/extended/builds/start.go

```diff
@@ -51,6 +51,25 @@ var _ = g.Describe("[builds][Slow] starting a build using CLI", func() {
 		})
 	})
 
+	g.Describe("setting OwnerReferences", func() {
+		g.It("should set parent BuildConfig as OwnerReference", func() {
+			g.By("starting the build from buildconfig")
+			br, err := exutil.StartBuildAndWait(oc, "sample-build")
```

Comment:
Given that there's no need for the build to run, and factoring in my suggestions above, can't this all be done in existing unit tests in generator_test.go?  Search there for references to generateBuildFromConfig and generateBuildFromBuild.